packages/rum-core/src/domain/view/viewCollection.ts

```diff
@@ -77,26 +77,26 @@ function processViewUpdate(
-      cumulative_layout_shift: view.commonViewMetrics.cumulativeLayoutShift,
-      cumulative_layout_shift_target_selector: view.commonViewMetrics.cumulativeLayoutShiftTargetSelector,
-      first_byte: toServerDuration(view.initialViewMetrics.firstByte),
-      dom_complete: toServerDuration(view.initialViewMetrics.domComplete),
-      dom_content_loaded: toServerDuration(view.initialViewMetrics.domContentLoaded),
-      dom_interactive: toServerDuration(view.initialViewMetrics.domInteractive),
+      cumulative_layout_shift: view.commonViewMetrics.cumulativeLayoutShift?.value,
+      cumulative_layout_shift_target_selector: view.commonViewMetrics.cumulativeLayoutShift?.targetSelector,
+      first_byte: toServerDuration(view.initialViewMetrics.navigationTimings?.firstByte),
+      dom_complete: toServerDuration(view.initialViewMetrics.navigationTimings?.domComplete),
+      dom_content_loaded: toServerDuration(view.initialViewMetrics.navigationTimings?.domContentLoaded),
+      dom_interactive: toServerDuration(view.initialViewMetrics.navigationTimings?.domInteractive),
       error: {
         count: view.eventCounts.errorCount,
       },
       first_contentful_paint: toServerDuration(view.initialViewMetrics.firstContentfulPaint),
-      first_input_delay: toServerDuration(view.initialViewMetrics.firstInputDelay),
-      first_input_time: toServerDuration(view.initialViewMetrics.firstInputTime),
-      first_input_target_selector: view.initialViewMetrics.firstInputTargetSelector,
-      interaction_to_next_paint: toServerDuration(view.commonViewMetrics.interactionToNextPaint),
-      interaction_to_next_paint_target_selector: view.commonViewMetrics.interactionToNextPaintTargetSelector,
+      first_input_delay: toServerDuration(view.initialViewMetrics.firstInputTimings?.firstInputDelay),
+      first_input_time: toServerDuration(view.initialViewMetrics.firstInputTimings?.firstInputTime),
+      first_input_target_selector: view.initialViewMetrics.firstInputTimings?.firstInputTargetSelector,
+      interaction_to_next_paint: toServerDuration(view.commonViewMetrics.interactionToNextPaint?.value),
+      interaction_to_next_paint_target_selector: view.commonViewMetrics.interactionToNextPaint?.targetSelector,
       is_active: view.isActive,
       name: view.name,
-      largest_contentful_paint: toServerDuration(view.initialViewMetrics.largestContentfulPaint),
-      largest_contentful_paint_target_selector: view.initialViewMetrics.largestContentfulPaintTargetSelector,
-      load_event: toServerDuration(view.initialViewMetrics.loadEvent),
+      largest_contentful_paint: toServerDuration(view.initialViewMetrics.largestContentfulPaint?.value),
+      largest_contentful_paint_target_selector: view.initialViewMetrics.largestContentfulPaint?.targetSelector,
+      load_event: toServerDuration(view.initialViewMetrics.navigationTimings?.loadEvent),
       loading_time: discardNegativeDuration(toServerDuration(view.commonViewMetrics.loadingTime)),
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**💭 thought:** ‏this is a *lot* of unminifiable code

Reply:
Yes. Do you have something in mind to improve it?

Reply:
One way could be to change the intermediary representation so we can use `combine` and avoid repeating every single fields. For example, `InitialViewMetrics` could be:

```ts
export interface InitialViewMetrics {
  first_contentful_paint?: ServerDuration
  first_byte?: ServerDuration
  ...
}
```

Reply:
Since this PR is rather large already, we might want to postpone this change after thinking a bit more about it.